src/components/DataTable.tsx

```diff
@@ -571,7 +571,21 @@ export function DataTable<T extends object>(props: Props<T>) {
       setCurrentPage(1);
     }
 
-    setAreRowsRendered(true);
+    if (perPage === '10') {
+      setAreRowsRendered(true);
+    }
+
+    if (perPage === '50') {
+      setTimeout(() => {
+        setAreRowsRendered(true);
+      }, 50);
+    }
+
+    if (perPage === '100') {
+      setTimeout(() => {
+        setAreRowsRendered(true);
+      }, 150);
```

Comment:
@beganovich Very small timeouts are added just for the purpose of UX improvement. I removed them in one of the previous versions, but I see right now that we will need just a small one. So, basically the time that is taken for rendering table rows - it is only needed when per page is 50 or 100.